src/plugins/unified_search/public/filters_builder/filters_builder_utils.ts

```diff
@@ -0,0 +1,344 @@
+      targetArray.splice(selector, 1, buildOrFilter([targetArray[selector], filter]));
+    }
+    if (conditionalType === ConditionTypes.AND) {
+      targetArray.splice(selector, 1, [targetArray[selector], filter]);
+    }
+  } else {
+    targetArray.splice(selector + 1, 0, filter);
+  }
+
+  return newFilters;
+};
+
+export const removeFilter = (filters: Filter[], path: string) => {
+  const newFilters = cloneDeep(filters);
+  const pathInArray = getPathInArray(path);
+  const { targetArray } = getContainerMetaByPath(newFilters, pathInArray);
+  const selector = pathInArray[pathInArray.length - 1];
+
+  targetArray.splice(selector, 1);
+
+  return normalizeFilters(newFilters);
+};
+
+const addFilterThenRemoveFilter = (
+  filters: Filter[],
+  filter: FilterItem,
+  from: string,
+  to: string,
+  conditionalType: ConditionTypes
+) => {
+  const newFiltersWithFilter = addFilter(filters, filter, to, conditionalType);
+  return removeFilter(newFiltersWithFilter, from);
+};
+
+const removeFilterThenAddFilter = (
+  filters: Filter[],
+  filter: FilterItem,
+  from: string,
+  to: string,
+  conditionalType: ConditionTypes
+) => {
+  const newFiltersWithoutFilter = removeFilter(filters, from);
+  return addFilter(newFiltersWithoutFilter, filter, to, conditionalType);
+};
+
+export const moveFilter = (
+  filters: Filter[],
+  from: string,
+  to: string,
+  conditionalType: ConditionTypes
+) => {
+  const newFilters = cloneDeep(filters);
+  const movingFilter = getFilterByPath(newFilters, from);
+
+  const pathInArrayTo = getPathInArray(to);
+  const pathInArrayFrom = getPathInArray(from);
+
+  if (pathInArrayTo.length === pathInArrayFrom.length) {
+    const filterPositionTo = pathInArrayTo.at(-1);
+    const filterPositionFrom = pathInArrayFrom.at(-1);
+
+    const { parentConditionType } = getContainerMetaByPath(newFilters, pathInArrayTo);
+    const filterMovementDirection = Number(filterPositionTo) - Number(filterPositionFrom);
+
+    if (filterMovementDirection === -1 && parentConditionType === conditionalType) {
+      return filters;
+    }
+
+    if (filterMovementDirection >= -1) {
+      return addFilterThenRemoveFilter(newFilters, movingFilter, from, to, conditionalType);
+    } else {
+      return removeFilterThenAddFilter(newFilters, movingFilter, from, to, conditionalType);
+    }
+  }
+
+  if (pathInArrayTo.length > pathInArrayFrom.length) {
+    return addFilterThenRemoveFilter(newFilters, movingFilter, from, to, conditionalType);
+  } else {
+    return removeFilterThenAddFilter(newFilters, movingFilter, from, to, conditionalType);
+  }
+};
+
+export const updateFilter = (
+  filters: Filter[],
+  path: string,
+  field?: DataViewField,
+  operator?: Operator,
+  params?: Filter['meta']['params']
+) => {
+  const newFilters = [...filters];
+  const changedFilter = getFilterByPath(newFilters, path) as Filter;
+  let filter = Object.assign({}, changedFilter);
+
+  if (field && operator && params) {
```

Comment:
@nlatipov  what do you think about some renaming 

```
updateArrayParams -> updateWithIsOneOfOperator
updateParams -> updateWithIsOperator
updateOperatorWithExistsType -> updateWithExistsOperator
updateOperator -> remove. Looks as a duplicate of updateParams
```

Reply:
I think it's a good idea